lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -41,6 +54,36 @@ def process_root_folder
                 end
             end
 
+            # Manages folder available space
+            #
+            # The method will check if there is enough space to save more log files
+            # according to pre-established threshold.
+            #
+            # @param [integer] free_space_low_limit: required free space threshold, at
+            #   which the archiver starts deleting the oldest log files
+            # @param [integer] free_space_delete_until: post-deletion free space, at which
+            #   the archiver stops deleting the oldest log files
+            def ensure_free_space(free_space_low_limit: @free_space_low_limit,
+                free_space_delete_until: @free_space_delete_until)
+                stat = Sys::Filesystem.stat(@target_dir)
+                available_space = stat.bytes_free
+
+                return if available_space > free_space_low_limit
+
+                until available_space >= free_space_delete_until
+                    files = @target_dir.each_child.select(&:file?)
+                    break if files.empty?
+
+                    available_space += files.sort.first.size
+                    files.sort.first.unlink
```

Comment:
Do not duplicate long method calls. Give them names (plus, you are actually **sorting** the array twice ... quite wasteful)

```
removed_file = files.sort.first
removed_file.unlink
available_space += removed_file.size
```

Reply:
(the inversion of unlink and available_space is to update available_space only after we actually removed the file ... does not change the correctness of the method, just something that I would do personally)